solutions/cf-862B.mdx

```diff
@@ -0,0 +1,123 @@
+---
+id: cf-862B
+source: CF
+title: Bipartiteness
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/54590) 
+
+## Explanation
+Given the number of colored and uncolored nodes, the maximum number of edges that can exist is $c \cdot uc$ where $c$ represents the number of colored nodes, and $uc$ represents the number of uncolored nodes. 
+
+This is because an edge can be drawn from each colored node to every other uncolored node. However, we already have $n - 1$ edges, so the answer is $(c \cdot uc) - (n - 1)$.
+
+To find the number of colored and uncolored nodes, we can run a [DFS](https://usaco.guide/silver/dfs), incrementing the number of colored and uncolored nodes as we traverse the graph.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Bipartiteness {
+	public static ArrayList<Integer>[] adj;
+	
+	// counters for colored and uncolored nodes
+	public static long[] count = {0, 0};
+
+	public static void main(String[] args) throws IOException{
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();
+
+		// +1 for 1-indexed nodes
+		adj = new ArrayList[n + 1];
+		
+		for (int i = 0; i <= n; i++) {
+			adj[i] = new ArrayList<>();
+		}
+
+		for (int i = 0; i < n - 1; i++) {
+			int u = io.nextInt();
+			int v = io.nextInt();
+			
+			adj[u].add(v);
+			adj[v].add(u);
+		}
+		
+		dfs(1, 0, false);
+		
+		// ans = (all possible edges) - (existing edges)
+		long ans = (count[0] * count[1]) - (n - 1); 
+		
+		io.println(ans);
+		io.close();
+	}
+
+	public static void dfs(int node, int parent, boolean color) {
+		count[color ? 0 : 1]++;
```

Comment:
Use if statement here
```suggestion
		if (color) {
				count[0]++;
		}
		else {
				count[1]++;
		}
```

Reply:
dustin since when was a simple ternary statement so scary

Reply:
I feel like this is something you would do

Reply:
i mean this one is really simple, i only ask people to unwrap ungodly one-liners

Reply:
also why are the indents two tabs?

Reply:
err thats my bad i suck at github